src/main/java/org/jenkinsci/plugins/orgfolder/github/MainLogic.java

```diff
@@ -162,24 +162,26 @@ public void applyRepo(WorkflowMultiBranchProject item, GitHubSCMNavigator scm) t
      */
     public void applyBranch(WorkflowJob branch, WorkflowMultiBranchProject repo, GitHubSCMNavigator scm) throws IOException {
         if (UPDATING.get().add(branch)) {
-            Branch b = repo.getProjectFactory().getBranch(branch);
-            GitHubLink repoLink = repo.getAction(GitHubLink.class);
-            if (repoLink!=null) {
-                BulkChange bc = new BulkChange(branch);
-                try {
+            BulkChange bc = new BulkChange(branch);
+            try {
+                Branch b = repo.getProjectFactory().getBranch(branch);
+                GitHubLink repoLink = repo.getAction(GitHubLink.class);
+                if (repoLink!=null) {
                     ChangeRequestAction action = b.getHead().getAction(ChangeRequestAction.class);
                     String url;
                     if (action == null) {
+                        // branch in this repository
                         url = repoLink.getUrl() + "/tree/" + b.getName();
                     } else {
+                        // pull request to this repository
                         url = repoLink.getUrl() + "/pull/" + action.getId();
                     }
                     branch.replaceAction(new GitHubLink("branch", url));
                     bc.commit();
-                } finally {
-                    bc.abort();
-                    UPDATING.get().remove(branch);
                 }
+            } finally {
+                bc.abort();
+                UPDATING.get().remove(branch);
```

Comment:
Ditto here, two `finally` blocks would be safer.

BTW can we make sure `BulkChange` is compatible with `try`-with-resources?